sycl/include/sycl/detail/defines.hpp

```diff
@@ -40,8 +40,9 @@
 #endif
 
 // joint matrix should only be included by default for SPIR or NVPTX backends
-#if defined __SPIR__ || defined __NVPTX__ || !defined __SYCL_DEVICE_ONLY__
+#if defined __SPIR__ || defined __NVPTX__ || !defined __SYCL_DEVICE_ONLY__ ||  \
+    defined __gfx90a__
```

Comment:
I just had a thought: what happens if you compile on AMD backend but not `__gfx90a__` . Does the compiler gives a reasonable diagnostic to tell the user that joint_maitrx is only supported for `gfx90a` on the hip backend?

Reply:
I see: https://github.com/intel/llvm/pull/11485/files#r1358183311